x-pack/plugins/lens/public/xy_visualization/xy_config_panel/threshold_panel.tsx

```diff
@@ -291,35 +297,32 @@ const LineThicknessSlider = ({
   value: number;
   onChange: (value: number) => void;
 }) => {
-  const onChangeWrapped = useCallback(
-    (newValue) => {
-      if (Number.isInteger(newValue)) {
-        onChange(getSafeValue(newValue, newValue, minRange, maxRange));
-      }
-    },
-    [onChange]
-  );
-  const { inputValue, handleInputChange } = useDebouncedValue<number | ''>(
-    { value, onChange: onChangeWrapped },
-    { allowFalsyValue: true }
-  );
+  const [unsafeValue, setUnsafeValue] = useState<string>(String(value));
 
   return (
     <EuiRange
       fullWidth
       data-test-subj="lnsXY_lineThickness"
-      value={inputValue}
+      value={unsafeValue}
       showInput
       min={minRange}
       max={maxRange}
       step={1}
       compressed
-      onChange={(e) => {
-        const newValue = e.currentTarget.value;
-        handleInputChange(newValue === '' ? '' : Number(newValue));
+      onChange={({ currentTarget: { value: newValue } }) => {
+        setUnsafeValue(newValue);
+        const convertedValue = newValue === '' ? '' : Number(newValue);
+        const safeValue = getSafeValue(Number(newValue), Number(newValue), minRange, maxRange);
+        // only update onChange is the value is valid and in range
+        if (convertedValue === safeValue) {
+          onChange(safeValue);
+        }
       }}
-      onBlur={() => {
-        handleInputChange(getSafeValue(inputValue, value, minRange, maxRange));
+      onBlur={({ currentTarget: { value: newValue } }) => {
+        const convertedValue = newValue === '' ? '' : Number(newValue);
+        if (Number.isInteger(convertedValue)) {
+          onChange(getSafeValue(convertedValue, Number(convertedValue), minRange, maxRange));
+        }
```

Comment:
The `onBlur` callback is not working as expected: then tabbing from the input empty to the next field the previous value is not shown and the input remains empty.

This is a fix for that, but there's another problem upstream in the `useDebouceValue`:

```suggestion
      onBlur={() => {
        if (unsafeValue !== String(value)) {
          onChange(
            getSafeValue(
              unsafeValue === '' ? unsafeValue : Number(unsafeValue),
              value,
              minRange,
              maxRange
            )
          );
        }
      }}
```